src/java/com/google/cloud/bigquery/dwhassessment/hooks/MigrationAssessmentLoggingHook.java

```diff
@@ -0,0 +1,130 @@
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(MigrationAssessmentLoggingHook.class.getName());
+
+  private static final Schema QUERY_EVENT_SCHEMA = AvroSchemaLoader.loadSchema("QueryEvents.avsc");
+
+  // Just for the example. Real hook will write it to the file
+  public ArrayList<GenericRecord> records = new ArrayList<>();
+
+  public void run(HookContext hookContext) throws Exception {
+    processQueryEventAvro(hookContext);
+  }
+
+  private void processQueryEventAvro(HookContext hookContext) {
+    QueryPlan plan = hookContext.getQueryPlan();
+
+    LOG.info("Received hook notification for: " + plan.getQueryId());
+
+    // Make a copy so that we do not modify hookContext conf.
+    HiveConf conf = new HiveConf(hookContext.getConf());
+    List<ExecDriver> mrTasks = Utilities.getMRTasks(plan.getRootTasks());
+    List<TezTask> tezTasks = Utilities.getTezTasks(plan.getRootTasks());
+    ExecutionMode executionMode = getExecutionMode(mrTasks, tezTasks);
+
+    GenericRecord queryEvent =
+        new GenericRecordBuilder(QUERY_EVENT_SCHEMA)
+            .set("QueryId", plan.getQueryId())
+            .set("QueryText", plan.getQueryStr())
+            .set("EventType", hookContext.getHookType().name())
+            .set("Timestamp", plan.getQueryStartTime())
+            .set("User", getUser(hookContext))
+            .set("RequestUser", getRequestUser(hookContext))
+            .set("ExecutionMode", executionMode.name())
+            .set("Queue", getQueueName(executionMode, conf))
+            .set("TablesRead", getTablesFromEntitySet(plan.getInputs()))
+            .set("TablesWritten", getTablesFromEntitySet(plan.getOutputs()))
+            .build();
+    records.add(queryEvent);
+    LOG.info("Processed record: " + queryEvent.toString());
+  }
+
+  private List<String> getTablesFromEntitySet(Set<? extends Entity> entities) {
+    List<String> tableNames = new ArrayList<>();
+    for (Entity entity : entities) {
+      if (entity.getType() == TABLE || entity.getType() == PARTITION) {
+        tableNames.add(entity.getTable().getDbName() + "." + entity.getTable().getTableName());
+      }
+    }
+    return tableNames;
+  }
+
+  private String getUser(HookContext hookContext) {
+    return hookContext.getUgi().getShortUserName();
+  }
+
+  private String getRequestUser(HookContext hookContext) {
+    String requestUser = hookContext.getUserName();
+    return requestUser == null ? hookContext.getUgi().getUserName() : requestUser;
+  }
+
+  private ExecutionMode getExecutionMode(List<ExecDriver> mrTasks, List<TezTask> tezTasks) {
+    if (tezTasks.size() > 0) {
+      // Need to go in and check if any of the tasks is running in LLAP mode.
+      for (TezTask tezTask : tezTasks) {
+        if (tezTask.getWork().getLlapMode()) {
+          return ExecutionMode.LLAP;
+        }
+      }
+      return ExecutionMode.TEZ;
+    } else if (mrTasks.size() > 0) {
```

Comment:
nit: There is no need to use `else`, if the body of the `if` instruction above ends with `return`. Similarly 2 lines below.

Reply:
Simplified it, done.